promql/functions.go

```diff
@@ -814,11 +863,11 @@ var functions = map[string]*Function{
 		Call:       funcResets,
 	},
 	"round": {
-		Name:         "round",
-		ArgTypes:     []model.ValueType{model.ValVector, model.ValScalar},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcRound,
+		Name:       "round",
+		ArgTypes:   []model.ValueType{model.ValVector, model.ValScalar},
+		Variadic:   true,
```

Comment:
As a callee I have every freedom to ignore arguments or parts of them.

Ultimately, what you want to achieve is a static error on more than one variadic argument – that's the exact opposite of Python, which you name as an example for a use of that feature. Those functions can still throw runtime errors if that's what you are worried about.

Our expression language is statically typed in general, so scripting languages are not a fair comparison. I don't think it needs a type system more complex than that of an actual programming language.